File: arrays/two_sum/two_sum.js
const twoSum = function(nums, target) {
  const prev = new Map();
  for (let i = 0; i < nums.length; i++) {
      if (prev.has(target - nums[i])) {
          return [prev.get(target-nums[i]), i];
      }
      prev.set(nums[i], i);
  }
  return []
};

File: arrays/two_sum/two_sum.cc
#include <vector>
#include <unordered_map>
using namespace std;

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        unordered_map<int, int> prev;
        vector<int> result;
        for (int i=0; i<nums.size(); i++) {
            if (prev.find(target-nums[i]) != prev.end()) {
                result = {prev[target-nums[i]], i};
                return result;
            }
            prev[nums[i]] = i;
        }
        result = {0, 1};
        return result;
    }
};